### DISCUSSION TITLE
FluidSynth on Daisy DSP?

### DISCUSSION THREAD

**Discussion by GeoffreyPlitt** [FluidSynth/fluidsynth#1100] [General]
Anybody seen these? https://www.electro-smith.com/daisy

How cool would it be to run FluidSynth on the Daisy Petal, Patch, or Field?

The guts is a system-on-a-chip with some powerful DSP, but I'm not sure if it can run general OS's, might be a more embedded environment.

**Comment by znmeb** [comment]
I've got two of the Daisy units - a Pod and a Field. They are indeed embedded. The native tooling is either C++ code or imports of Pure Data / MAX GEN patches. There's also a way to use DSP code written in Faust. The code is mostly open source.

The getting started pages are here: <https://github.com/electro-smith/DaisyWiki/wiki>. I doubt if you'd be able to get FluidSynth to work; it needs a lot of OS-level libraries that don't exist in the embedded world, and there is a constraint on flash size. I'd recommend viewing the Daisy intro videos on YouTube if you want to buy one of them and make music: <https://youtu.be/e4KaBs6qSkU>

**Comment by GeoffreyPlitt** [comment]
I'd love if FluidSynth had a port that traded those OS system calls for something more statically linked, but I know it's a lot of work.  I see https://github.com/divideconcept/FluidLite but I'm not sure if it's active.

Are there any other Soundfont-based audio engines that run on embedded systems?

**Reply by znmeb** [reply]
> I'd love if FluidSynth had a port that traded those OS system calls for something more statically linked, but I know it's a lot of work. I see https://github.com/divideconcept/FluidLite but I'm not sure if it's active.
> 
> Are there any other Soundfont-based audio engines that run on embedded systems?

I don't know of any, but I'm neither an embedded expert or a Soundfont expert. The Daisy hardware is interesting; the audio buffer management has dedicated resources onboard, which cuts down the load on the microcontroller.
